xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/classes/StaticListClassPropertyValuesProvider.java

```diff
@@ -49,13 +51,64 @@
 
     @Override
     protected PropertyValues getAllowedValues(StaticListClass propertyDefinition, int limit, String filter)
-        throws Exception
     {
         final PropertyValues result = new PropertyValues();
         List<String> allValues = StaticListClass.getListFromString(propertyDefinition.getValues());
+        Map<String, ListItem> valueMap = StaticListClass.getMapFromString(propertyDefinition.getValues());
+
+        String lowerFilter = filter.toLowerCase();
+
+        allValues.stream().map(id -> constructPropertyValueForId(id, propertyDefinition, valueMap))
+            // Filter both by key and by (translated) label (that's why we construct a PropertyValue first)
+            .filter(val -> val.getValue().toString().toLowerCase().contains(lowerFilter)
```

Comment:
You could have used `StringUtils#containsIgnoreCase` to simplify it.

Reply:
Didn't know about that (and it hasn't been used before), I've changed the code to use it.